src/App.js

```diff
@@ -25,13 +40,46 @@ export default class App extends Component {
     localStorage.setItem('todos', JSON.stringify(todo));
   }
 
+  onHandleCheckBoxChange = (id) => {
+    const todos = JSON.parse(localStorage.getItem('todos'));
+    todos.forEach((todo) => {
+      if (todo.id === id) {
+        todo.complete = !todo.complete;
+      }
+    });
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }
+
+  onDeleteHandler = (id) => {
+    const todos = JSON.parse(localStorage.getItem('todos'));
+    let selectedTodo = null;
+    todos.forEach((item) => {
+      if (item.id === id) {
+        selectedTodo = item;
+      }
+    });
+    if (selectedTodo.complete) {
```

Comment:
The user should have the ability to remove a task even if it wasn't completed. To achieve that you can remove these lines.